node-game2d-lib/index.js

```diff
@@ -0,0 +1,258 @@
+
+'use strict'
+
+/**
+ * Make the value -1 if less than 0 or 1 if more than 0.
+ */
+module.exports.sign = (value) => (parseFloat(value) < 0.0) ? -1.0 : 1.0;
+
+/**
+ * make a value be between the minimum and maximum
+ */
+module.exports.clamp = (value, minimum, maximum) =>
+{
+  let min = parseFloat(minimum);
+  let max = parseFloat(maximum);
+  let resut = parseFloat(value);
+  if (min > max) 
+  {
+    let temp = min;
+    min = max;
+    max = temp;
+  }
+
+  if (resut < min) 
+  {
+    resut = min;
+  }
+  else if (resut > max) 
+  {
+    resut = max;
+  }
+  return resut;
+}
+
+// Maybe removing the result variable and using the value itself can be better
+module.exports.clamp01 = (value) => 
+{
+  let num = parseFloat(value);
+  let result = 0.0;
+  if (num < 0.0) 
+  {
+    result = 0.0;
+  }
+  else if (num > 1.0) 
+  {
+    result = 1.0;
+  }
+  else 
+  {
+    result = num;
+  }
+  return result;
+}
+
+// I need to make the the VectorA and VectorB has x and y properties
+module.exports.distance = (vectorA, vectorB) => 
+{
+  let x = vectorB.x - vectorA.x
+  let y = vectorB.y - vectorA.y;
+  let result = Math.sqrt((x * x) + (y * y));
+  console.log("Distance from " + vectorA + " to " + vectorB + " is " + result)
+  return result
+}
+
+// I need to make the the VectorA and VectorB has x and y properties
+module.exports.squareDistance = (vectorA, vectorB) => 
+{
+  let x = vectorB.x - vectorA.x
+  let y = vectorB.y - vectorA.y;
+  let result = (x * x) + (y * y);
+  console.log("Square Distance from " + vectorA + " to " + vectorB + " is " + result)
+  return result;
+}
+
+// I need to make the the Vector has x and y properties
```

Comment:
I like that you have put comments on the code. It would be useful if you could also add a little more explanation about what the specific function is doing in further detail.

Reply:
It kinda goes over the scope. I put too many functions and didn't have the proper time to comment or test them all. In the future, more comments will come. Thanks